File: benchmarks/base64-encode/main.php
<?php

$i1 = "Hello, World!";
$i2 = "The quick brown fox jumps over the lazy dog.";
$i3 = "AaBbCcDdEeFf123456_!@#&*^()$=+-";

$before = hrtime(true);

base64_encode($i1);
base64_encode($i2);
base64_encode($i3);

$after = hrtime(true);

echo $after - $before;

File: benchmarks/base64-encode/main.cpp
#include <iostream>
#include <chrono>
#include <string>

#include "../../lib/cpp/base64/include/base64.hpp"

int main() {
    const auto before = std::chrono::steady_clock::now();

    std::string i1 = "Hello, World!";
    std::string i2 = "The quick brown fox jumps over the lazy dog.";
    std::string i3 = "AaBbCcDdEeFf123456_!@#&*^()$=+-";

    base64::to_base64(i1);
    base64::to_base64(i2);
    base64::to_base64(i3);

    const auto after = std::chrono::steady_clock::now();
    const auto duration = after - before;

    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(duration).count() << std::endl;

    return 0;
}